pkg/zonemgr/zone_file.go

```diff
@@ -2,180 +2,151 @@ package zonemgr
 	return &ZoneFile{
-		clusterName:        clusterName,
-		userAdminEmail:     userAdminEmail,
-		userNameserverName: userNameserverName,
-		userNameserverIP:   userNameserverIP,
-		userSubdomain:      userSubdomain,
+		nameServerIP: nameServerIP,
+		domain:       domain,
 	}
 }
 
 func (zoneFile *ZoneFile) init() {
-	zoneFile.setDefaultValues()
+	zoneFile.initCustomFields()
 	//before SOA serial
 	zoneFile.generateHeaderPrefix()
 	//after SOA serial
 	zoneFile.generateHeaderSuffix()
 	zoneFile.soaSerial = 0 // TODO make default RANDOM: time.Now().UnixNano()
 	zoneFile.header = zoneFile.generateHeader()
 	zoneFile.content = zoneFile.header
-	zoneFile.vmiRecordsMap = make(map[string]map[string]ARecord)
+	zoneFile.vmiRecordsMap = make(map[string][]string)
 }
 
-func (zoneFile *ZoneFile) setDefaultValues() {
-	zoneFile.domainName = fmt.Sprintf("%s.secondary", zoneFile.clusterName)
-
-	if zoneFile.userSubdomain != "" {
-		zoneFile.domainName = fmt.Sprintf("%s.%s", zoneFile.domainName, zoneFile.userSubdomain)
-	}
-
-	if zoneFile.userNameserverName == "" {
-		zoneFile.userNameserverName = fmt.Sprintf("ns1.%s", zoneFile.domainName)
-	}
-	/*
-		if zoneFile.userNameserverIP == "" {
-			//TODO default???
-		}
-	*/
-	if zoneFile.userAdminEmail == "" {
-		zoneFile.userAdminEmail = fmt.Sprintf("admin.%s", zoneFile.domainName)
+func (zoneFile *ZoneFile) initCustomFields() {
+	if zoneFile.domain == "" {
+		zoneFile.domain = domainDefault
+	} else {
+		zoneFile.domain = fmt.Sprintf("%s.%s", domainDefault, zoneFile.domain)
 	}
+	zoneFile.nameServerName = fmt.Sprintf("%s.%s", nameServerDefault, zoneFile.domain)
+	zoneFile.adminEmail = fmt.Sprintf("%s.%s", adminEmailDefault, zoneFile.domain)
 }
 
-/*
-$ORIGIN mycluster.secondary.mydomain.com
-$TTL 1750
-@       IN      SOA ns1.mycluster.secondary.mydomain.com admin@secdns.com (
-*/
 func (zoneFile *ZoneFile) generateHeaderPrefix() {
-	zoneFile.headerPref = fmt.Sprintf("\n$ORIGIN %s \n$TTL %s \n@ IN SOA %s %s (", zoneFile.domainName, ttl,
-		zoneFile.userNameserverName, zoneFile.userAdminEmail)
-
-	//fmt.Println("======= prefix:\n", zoneFile.headerPref)
-
+	zoneFile.headerPref = fmt.Sprintf("$ORIGIN %s. \n$TTL %s \n@ IN SOA %s. %s. (", zoneFile.domain, ttl,
+		zoneFile.nameServerName, zoneFile.adminEmail)
 }
 
-/*
-									7200       ; refresh (2 hours)
-	                                3600       ; retry (1 hour)
-	                                1209600    ; expire (2 weeks)
-	                                3600       ; minimum (1 hour)
-	                                )
-	        IN      NS  ns1.mycluster.secondary.mydomain.com
-	        IN      A   1.2.3.4
-*/
 func (zoneFile *ZoneFile) generateHeaderSuffix() {
 	zoneFile.headerSuf = fmt.Sprintf(" %s %s %s %s)\n", refresh, retry, expire, ttl)
 
-	if zoneFile.userNameserverName != "" {
-		zoneFile.headerSuf += fmt.Sprintf("IN NS %s\n", zoneFile.userNameserverName)
+	if zoneFile.nameServerName != "" {
```

Comment:
it will always be not empty, because you update it above according defaults
but it is optional isn't it ?
so if the os.getenv is empty we shouldn't include it IIUC ?

Reply:
os.getenv() will always be empty as we decided to go with hardcoded default value at this stage.
And we might need this IN NS record along with IN A record to describe user nameserver.
So I'll base the condition of adding both of those records on os.getenv(NAME_SERVER_IP) value.
If it exist then both NS and A records will be added.

Reply:
i mean this line will make `nameServerName` not empty
https://github.com/kubevirt/kubesecondarydns/pull/24/files#diff-25b5f948c0012399238bc50518ea6bc1f350c663cffcfd9f5c41d4e026ebf671R65
which happens anyhow, not based on the os.getenv